src/org/mozilla/javascript/VMBridge.java

```diff
@@ -88,22 +85,50 @@ private static VMBridge makeInstance()
      *
      * @param interfaces Array with one or more interface class objects.
      */
-    protected abstract Object getInterfaceProxyHelper(ContextFactory cf,
-                                             Class<?>[] interfaces);
+    protected abstract Object getInterfaceProxyHelper(ContextFactory cf, Class<?>[] interfaces);
 
     /**
-     * Create proxy object for {@link InterfaceAdapter}. The proxy should call
-     * {@link InterfaceAdapter#invoke(ContextFactory, Object, Scriptable,
-     *                                Object, Method, Object[])}
-     * as implementation of interface methods associated with
-     * <code>proxyHelper</code>. {@link Method}
+     * Create proxy object for {@link InterfaceAdapter}. The proxy should call {@link
+     * InterfaceAdapter#invoke(ContextFactory, Object, Scriptable, Object, Method, Object[])} as
+     * implementation of interface methods associated with <code>proxyHelper</code>. {@link Method}
      *
-     * @param proxyHelper The result of the previous call to
-     *        {@link #getInterfaceProxyHelper(ContextFactory, Class[])}.
+     * @param proxyHelper The result of the previous call to {@link
+     *     #getInterfaceProxyHelper(ContextFactory, Class[])}.
      */
-    protected abstract Object newInterfaceProxy(Object proxyHelper,
-                                       ContextFactory cf,
-                                       InterfaceAdapter adapter,
-                                       Object target,
-                                       Scriptable topScope);
+    protected abstract Object newInterfaceProxy(
+            Object proxyHelper,
+            ContextFactory cf,
+            InterfaceAdapter adapter,
+            Object target,
+            Scriptable topScope);
+
+    public abstract void discoverPublicMethods(Class<?> clazz, Map<MethodSignature, Method> map);
+
+    public static final class MethodSignature {
```

Comment:
`MethodSignature` is being used only to satisfy generics in a `Map<MethodSignature, Method>` but otherwise is used just as an `Object`. Here, it becomes `public` and hence part of the public API. That's one of the reasons why I kept this stuff as a package-visible subclass to `JavaMembers` (and @gbrail mentioned something along those lines when merging).

The current `master` code does not change the public API of Rhino, and would leave it as it is for 1.7.14. Once that release is out, the `VMBridge` API could be changed if there was an obvious gain from it, like trying to address non-public reflective accesses. I have in mind something along those lines although my approach would keep `JavaMembers_jdk11` as a class.